pkg/ride/functions_strings_test.go

```diff
@@ -54,6 +54,8 @@ func TestTakeString(t *testing.T) {
 		{[]rideType{rideInt(1)}, true, nil},
 		{[]rideType{}, true, nil},
 		{[]rideType{rideString("DRAGORION : Cradle of Many Strings\n[MYTHIC]🔶🔶🔶🔶🔶\n\nCeli, child of the first light. One of the main characters of the story, she is the first to see the vision of Cloudscape and its inhabitants from the Earth's dimension after the great destruction.\n\nDragorion - avatars sung into being by Eneria to bring sleep to the people of Cloudscape. They speak in dreams as lullabies, symphonies, hymns, arias and melodies. ~Legendarium\n\n©️Art of Monztre\n"), rideInt(50)}, false, rideString("DRAGORION : Cradle of Many Strings\n[MYTHIC]🔶🔶🔶?")},
+		// scala tests from https://github.com/wavesplatform/Waves/pull/3367
+		{[]rideType{rideString("x冬x"), rideInt(2)}, false, rideString("x冬")}, // the result is `x?` but it should be `x冬`
```

Comment:
this is where the test is not passed.
Expected :x冬
Actual   :x?
It's linked with https://github.com/wavesplatform/Waves/pull/3367/files#diff-8e87b4a107bf295f09ee6dbabd1d94780f554e2ed2276d3ffcad323436cf0534R2393